rewards/utils/emission_utils.py

```diff
@@ -32,6 +33,39 @@ def get_flat_emission_rate(sett: str, chain: str) -> Decimal:
     return Decimal(get_emission_control(chain).proRataEmissionRate(sett).call() / 1e4)
 
 
+@lru_cache
+def get_nft_control(chain: Network) -> ContractFunctions:
+    return make_contract(
+        EMISSIONS_CONTRACTS[chain]["NFTControl"], Abi.NFTControl, chain
+    )
+
+
+@lru_cache
```

Comment:
Same here, do you think we really need caching here?

Reply:
Well if we don't have caching here, then we will have to recompute the weights dict on each call which seems unnecessary. We could alter the structure of the code so that we remove get_nft_weight and we can just do those operations in the nft snapshot function